test/integration/compile-swagger-test.coffee

```diff
@@ -10,129 +10,140 @@ describe('compile() · Swagger', ->
   locationSchema = createLocationSchema()
 
   describe('causing a \'not specified in URI Template\' error', ->
-    errors = undefined
-    transactions = undefined
+    compilationResult = undefined
 
     beforeEach((done) ->
-      compileFixture(fixtures.notSpecifiedInUriTemplateAnnotation.swagger, (err, compilationResult) ->
-        return done(err) if err
-        {errors, transactions} = compilationResult
-        done()
+      compileFixture(fixtures.notSpecifiedInUriTemplateAnnotation.swagger, (args...) ->
+        [err, compilationResult] = args
+        done(err)
       )
     )
 
     it('is compiled into expected number of transactions', ->
-      assert.equal(transactions.length, 0)
+      assert.equal(compilationResult.transactions.length, 0)
+    )
+    it('is compiled with no warnings', ->
+      assert.deepEqual(compilationResult.warnings, [])
```

Comment:
I think it would be better to keep things consistent and use `assert.equal(array.length, expectedLength)` when checking for `expected number of <something>`.